### PR TITLE
Make things private

### PR DESCRIPTION
**Pull request by dentarg**
- [x] `#new` and `.internal_parse`
- [x] Constants

Close to #60.

### CODE REVIEW

**Comment by dentarg** [comment]
`rake profile:normalize` reports `Total: 1.791179` for `master` and `Total: 2.063033` for this branch

**Comment by dentarg** [comment]
screen shot 2015-12-11 at 20 11 56

**Comment by walro** [comment]
> rake profile:normalize reports Total: 1.791179 for master and Total: 2.063033 for this branch

I am guessing due to the extra three calls in `.parse` compared to `.internal_parse`. I think it's fine though.

LGTM!

**Comment by walro** [comment]
> I am guessing due to the extra three calls in .parse compared to .internal_parse. I think it's fine though.

Ah, no, I think we make an extra parsing now, I think we should fix that.

File: lib/twingly/url.rb [hunk 1]
```diff
@@ -101,7 +99,7 @@ def normalized
       normalized_url.host   = normalized_host
       normalized_url.path   = normalized_path
 
-      self.class.internal_parse(normalized_url)
+      self.class.parse(normalized_url)
```
Comment (walro) [inline lib/twingly/url.rb]: This will convert an addressable uri to a string and then back to a addressable uri. I think we want to avoid that.

Comment (walro) [inline lib/twingly/url.rb]: I think it's pretty easily amended by splitting `internal_parse` into two different ones, one for addressable and one for public suffix

Comment (walro) [inline lib/twingly/url.rb]: > I think it's pretty easily amended by splitting internal_parse into two different ones, one for addressable and one for public suffix

We already had a helper method apparently, I thought the change in 219cf37a3d53dfcd7a1c70e7560c508940eec4d6 made sense

**Comment by walro** [comment]
Would be interesting to re-run the benchmarks now

**Comment by dentarg** [comment]
> Would be interesting to re-run the benchmarks now

Total: 1.793219

**Comment by walro** [comment]
Cool

**Comment by dentarg** [comment]
@walro are we done? :)

**Comment by walro** [comment]
I think so!

**Pull request merged**